omod/src/test/java/org/openmrs/module/htmlformentryui/fragment/controller/htmlform/EnterHtmlFormFragmentControllerComponentTest.java

```diff
@@ -277,59 +291,6 @@ public void testSubmittingHtmlFormDefinedInUiResourceShouldAssociateWithExisting
 		assertThat(created.getEncounterDatetime(), is(visit.getStartDatetime())); // make sure the encounter date has been shifted to match the visit start time of 10:10:10
 	}
 	
-	@Test
-	public void testEditingHtmlFormDefinedInUiResourceShouldNotChangeTimeOfEncounterDateIfNewDateHasNoTimeComponentAndIsNotDifferentFromCurrentDate()
```

Comment:
Do you mind explaining why you completely removed this test?

Reply:
Hello @dkayiwa, before, on the encounterDate(showTime=true) we had 2 different widgets, date  + time. At the moment we have zonedDateTimeWidget, and it includes both, date and time, so it doesn't make sense to have a test without time, because this would give an error. That test is actually sending 00:00:00, and we consider that as midnight, and not as null.

Reply:
So because of those 2 things, 00:00:00 is not null and the new widget, i think that tests can be removed.

Reply:
@icrc-jfigueiredo how about those who have not yet migrated to the new zonedDateTimeWidget? Doesn't this test help to ensure that the widget they are using behaves as expected?

Reply:
zonedDateTimeWidget wont be retro compatible. If they are using encounterDate with `showTime=true` they will use the new widget with timezones. Or if they are using  `showTime=false` they are only using date without time, so this test is not for that case.

Reply:
@icrc-jfigueiredo is the switch to the new widget with timezones done automatically behind the scenes? Or will one have to modify the form definitions? Am referring to those who are using encounterDate with showTime=true

Reply:
For those who are using `encounterDate ` with `showTime=true`  now they will be forcing to use `zonedDateTimeWidget ` **but** to be able to use that `zonedDateTimeWidget `  we need to set the GP to true **or** the server and client must have the same TZ.

Reply:
Why are they forced to use `zonedDateTimeWidget` when their GP is set to `false`?

Reply:
They are not... We are only forced to use `zonedDateTimeWidget ` if we are using showTime=true

Reply:
If my GP is set to false, even if showTime=true, why not use the old widget?
In other words, why not use the new zonedDateTimeWidget only when the GP is set to true?
That way, we do not unintentionally introduce bugs for legacy forms which may be using showTime=true and do not care about the new timezones functionality.

Reply:
When we start developing timezones, one of the first things we talk, was about retro-compatibility, before timezones, if we had showTime=true we would have 2 widgets together, one for date and other for time, that was the way the things were done. With this new widget we break that functionality, we can no longer have the 2 widgets (date+time) together, now we have zonedDateTimeWidget, that brings Date + time to encounterDate. Obviously we can make a step back and bring back the old date&time widget but, i would be huge a step back. CC @mks-d

Reply:
Oh, i thought it was a mere if true then zonedDateTimeWidget else oldDateWidget.

Reply:
The "oldDateTimeWidget" no longer exists... Now or we use DateWidget or zonedDateTimeWidget

Reply:
> The "oldDateTimeWidget" no longer exists

Is it because we deleted it?

Reply:
Because it was calling Date Widget and Time widget on the `EncounterDetailSubmissionElement.java` we change that behaviour to call only `zonedDateTimeWidget`, we dont delete the "widget" we change they way things are instantiated.

Reply:
Can't we instantiate one or the other depending on the value of the GP?

Reply:
In theory yes, but we would need to do a lot of some changes on our code...

Reply:
I will let others weigh in. But personally, i find it safer using the new zonedDateTimeWidget only when one explicitly says so via setting the GP to true. Not because one already had `showTime=true`

Reply:
To bring DateTimeWidget back on encounterDate we need to change the behavior of EncounterDetailSubmissionElement

Reply:
Yeah, if we had already moved away from using the DateTie widget within the EncounterDetailSubmissionElement prior to your changes, I wasn't suggesting we bring it back just for that.